### PR TITLE
fix: extension performance issues 

### PR DESCRIPTION
**Pull request by AugmentedMode**
## Explanation

Fixes General extension perfomance issues

### `@metamask/package-a`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

### `@metamask/package-b`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [x] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/phishing-controller/src/utils.test.ts [hunk 1]
```diff
@@ -603,3 +604,47 @@ describe('getHostname', () => {
...
+
+describe('extractDomainName', () => {
+  it('should extract the primary domain from a standard hostname', () => {
+    const hostname = 'www.example.com';
+    const expected = 'example.com';
+    const result = extractDomainName(hostname);
+    expect(result).toBe(expected);
+  });
+
+  it('should extract the primary domain from a hostname with multiple subdomains', () => {
```
Comment (0xOhm) [inline packages/phishing-controller/src/utils.test.ts:616]: Should we add test cases for things like site.web.google.com? essentially testing for longer hostnames

File: packages/phishing-controller/src/utils.ts [hunk 2]
```diff
@@ -288,3 +288,66 @@ export const getHostnameFromUrl = (url: string): string | null => {
...
   return hostname;
 };
+
+/**
+ * Extracts the domain name (e.g., example.com) from a given hostname.
+ *
+ * @param hostname - The full hostname to extract the domain from.
+ * @returns The extracted domain name.
+ */
+export const extractDomainName = (hostname: string): string => {
```
Comment (mindofmar) [inline packages/phishing-controller/src/utils.ts:298]: This function is only used in tests, can it be moved into the test file?

File: packages/phishing-controller/src/utils.ts [hunk 3]
```diff
@@ -288,3 +288,66 @@ export const getHostnameFromUrl = (url: string): string | null => {
...
+
+  if (sourceParts.length === 0) {
+    return domains;
+  }
+
+  if (sourceParts.length === 1) {
+    // Single-segment hostname (e.g., 'uk')
+    domains.push(sourceParts[0].toLowerCase());
+  } else {
+    // Start with the base TLD (last two labels, e.g., 'co.uk')
```
Comment (mindofmar) [inline packages/phishing-controller/src/utils.ts:337]:
```suggestion
        // Start with the base domain or TLD (last two labels, e.g., 'co.uk' or 'example.com')
```

File: packages/phishing-controller/src/utils.ts [hunk 4]
```diff
@@ -288,3 +288,66 @@ export const getHostnameFromUrl = (url: string): string | null => {
...
+    const baseTLD = sourceParts.slice(-2).join('.');
```
Comment (mindofmar) [inline packages/phishing-controller/src/utils.ts:338]:
```suggestion
    const baseDomain = sourceParts.slice(-2).join('.');
```

**Review by mindofmar — CHANGES_REQUESTED** [review]

File: packages/phishing-controller/src/PhishingDetector.test.ts [hunk 5]
```diff
@@ -1298,6 +1299,33 @@ describe('PhishingDetector', () => {
...
     });
 
+    it('check the hash against c2DomainBlocklist, returning the correct result without a version with sub domains', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            c2DomainBlocklist: [
+              'a379a6f6eeafb9a55e378c118034e2751e682fab9f2d30ab13d2125586ce1947',
```
Comment (mindofmar) [inline packages/phishing-controller/src/PhishingDetector.test.ts:1309]:
```suggestion
              'a379a6f6eeafb9a55e378c118034e2751e682fab9f2d30ab13d2125586ce1947', // example.com
```

Just to improve readability of the test and to match the format of the other tests

**Review by mindofmar — APPROVED** [review]

**Review by 0xOhm — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**